ramanujan/enumerators/RelativeGCFEnumerator.py

```diff
@@ -0,0 +1,283 @@
+
+    Returns an integer with the matching key for the GCF, which is int(gcf_value*precision_factor) and the number of
+    digits calculated.
+    """
+    computed_values = []
+    items_computed = 0
+
+    # Some GCFs converge by oscillation. its better to take odd burst_number to test
+    # values from both sub-series
+    burst_number = burst_number if burst_number % 2 == 1 else burst_number + 1
+    next_gcf_calculation = burst_number if burst_number >= min_iters else min_iters
+
+    precision_factor = 10 ** result_precision
+    prev_q = 0
+    q = 1
+    prev_p = 1
+    # This is a ugly hack but it works. a[0] is handled before the rest here:
+    p = an_iterator.__next__()  # will place a[0] to p
+    bn_iterator.__next__()  # b0 is discarded
+
+    if p == 0:
+        raise ZeroInAn()
+
+    for i, (a_i, b_i) in enumerate(zip(an_iterator, bn_iterator)):
+        if a_i == 0:
+            raise ZeroInAn()
+
+        # a_i is the (i+1)'th item of an, and b_i the the i'th item of bn
+        tmp_a = q
+        tmp_b = p
+
+        q = a_i * q + b_i * prev_q
+        p = a_i * p + b_i * prev_p
+
+        prev_q = tmp_a
+        prev_p = tmp_b
+
+        if i == next_gcf_calculation:
+            next_gcf_calculation += burst_number * (len(computed_values) + 1)
+            if q != 0:  # safety check
+                computed_values.append(trunc_division(precision_factor * p, q))
+                items_computed += 1
+            else:
+                raise ZeroDivisionError()
+
+            if items_computed >= 2:
+                if computed_values[-1] == computed_values[-2]:
+                    return computed_values[-1], result_precision
+
+                if items_computed >= 3:
+                    # if the GCF oscillates between two sub-series - we expect the distance between the two to 
+                    # monotonically decrease. If the GCF doesn't oscillate, we expect the distance between two following
+                    # values to monotonically decrease as well. If none of those apply, we halt the calculation here.
+                    if abs(computed_values[-2] - computed_values[-1]) > abs(computed_values[-3] - computed_values[-2]):
+                        raise NotConverging("Not converging")
+
+    # The locations we calculate the GCF for and the size of the iterator is unrelated. Its likely that if we got here,
+    # that last values of the GCFs p and q were calculated but never got divided and added to computed_values
+    if i != next_gcf_calculation:
+        computed_values.append(trunc_division(precision_factor * p, q))
+        items_computed += 1
+
+    # we'll take the last two calculations, and check for matching digits.
+    # Once the two doesn't match, we'll know that we cannot trust the following digits.
+    res = ''
+    for i, (c1, c2) in enumerate(zip(str(computed_values[-2]), str(computed_values[-1]))):
```

Comment:
The number of matching digits can be calculated using bitwise functions if the digits are in binary:
```suggestion
    # bitwise xor leaves bits that aren't common and bit_length tells where they start
    non_common_bits = (computed_values[-2] ^ computed_values[-1]).bit_length()
    # preserve common bits, zero others (mask e.g. 11111000 for 3 non common bits)
    res = computed_values[-1] & (~((1 << non_common_bits) - 1))
    all_bits = max(computed_values[-2], computed_values[-1]).bit_length()
    common_bits = all_bits - non_common_bits
    return res, common_bits
```

Reply:
Looks like a cool idea, but on the rest of the code we're using digits and not bits to determine on precision, and using this method requires more changes. Ill look into it in the future! (we're looking to get this PR out soon)